src/main/java/com/google/planningworks/cdap/plugins/sink/AnaplanSinkConfig.java

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.planningworks.cdap.plugins.sink;
+
+import static com.anaplan.client.AnaplanService.NAME_MODEL_ID;
+import static com.anaplan.client.AnaplanService.NAME_SERVER_FILE_NAME;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import com.google.planningworks.cdap.plugins.base.AnaplanPluginConfig;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * The config class for {@link AnaplanSink} that contains all properties that need to be filled in
+ * by the user when building a pipeline.
+ */
+class AnaplanSinkConfig extends AnaplanPluginConfig {
+
+  public static final int CHUNK_SIZE = 10;
+  public static final Set<Schema.Type> SUPPORTED_TYPES =
+    ImmutableSet.of(Schema.Type.INT, Schema.Type.LONG, Schema.Type.STRING, Schema.Type.FLOAT,
+      Schema.Type.DOUBLE,
+      Schema.Type.BOOLEAN);
+  public static final Set<Schema.LogicalType> SUPPORTED_LOGICAL_TYPES =
+    ImmutableSet.of(Schema.LogicalType.DATE, Schema.LogicalType.DECIMAL);
+
+  @Name(NAME_SERVER_FILE_NAME)
+  @Macro
+  @Description("Anaplan Server file name")
+  private String serverFileName;
+
+  /**
+   * Validates the configure options entered by the user
+   *
+   * @param collector the shared failure info collector storing the validation failures
+   */
+  public void validate(FailureCollector collector, Schema schema) {
+    super.validate(collector);
+
+    if (!containsMacro(NAME_SERVER_FILE_NAME) && Strings.isNullOrEmpty(serverFileName)) {
+      collector
+        .addFailure("Server file is not presented.", null)
+        .withConfigProperty(NAME_MODEL_ID);
+    }
+
+    validateSchemaSupportedByAnaplan(schema, collector);
+  }
+
+  private void validateSchemaSupportedByAnaplan(Schema schema, FailureCollector collector) {
+    if (schema == null || schema.getFields() == null) {
+      return;
+    }
+
+    for (Schema.Field field : schema.getFields()) {
+      validateFieldSupportedByAnaplan(field, collector);
+    }
+  }
+
+  private static void validateFieldSupportedByAnaplan(Schema.Field field,
+    FailureCollector collector) {
+    Schema fieldSchema = getNonNullableSchema(field.getSchema());
+    Schema.Type type = fieldSchema.getType();
+    Schema.LogicalType logicalType = fieldSchema.getLogicalType();
+
+    // validate logical types
+    if (logicalType != null) {
+      if (!SUPPORTED_LOGICAL_TYPES.contains(logicalType)) {
+        collector.addFailure(generateUnsupportedTypeErrorMessage(field, fieldSchema),
```

Comment:
you can use 'withInputSchemaField()' to tell the UI to highlight the exact field that is causing problems. See https://github.com/data-integrations/google-cloud/blob/f0ae7de2be0d4798d771174c52aca19c4fb8ac0a/src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java#L262 for an example. 

This also lets you check more specific things in the unit test

Reply:
Yeah, that's more clear, updated!